pages/admin/user/[id].tsx

```diff
@@ -0,0 +1,308 @@
+import { joiResolver } from "@hookform/resolvers/joi";
+import { UpdateUserDTO } from "pages/api/admin/users/update";
+import { useForm } from "react-hook-form";
+
+interface AdminEditUserFormValues {
+  firstName: string;
+  lastName: string;
+  email: string;
+  phoneNumber?: string;
+  role: UserRole;
+}
+
+interface EditUserProps {
+  user?: User;
+  isEditing: boolean;
+  setIsEditing: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+type ModalProps = React.PropsWithChildren<{
+  open?: boolean;
+}>;
+
+const Modal: React.FC<ModalProps> = ({ children, open }: ModalProps) => {
+  return open ? (
+    <div className="absolute top-0 left-0 w-screen h-screen bg-dark bg-opacity-50 flex justify-center items-center">
+      <div className="bg-white rounded w-3/4 px-10 pt-12 pb-8">{children}</div>
+    </div>
+  ) : null;
+};
+
+const AdminEditUserFormSchema = Joi.object<AdminEditUserFormValues>({
+  firstName: Joi.string().trim().required(),
+  lastName: Joi.string().trim().required(),
+  email: Joi.string()
+    .trim()
+    .email({ tlds: { allow: false } })
+    .required(),
+  phoneNumber: Joi.string().optional(),
+  role: Joi.string()
+    .valid(...UserRoleConstants)
+    .required(),
+});
+
+const EditUser: React.FunctionComponent<EditUserProps> = ({
+  user,
+  isEditing,
+  setIsEditing,
+}) => {
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState("");
+  const { errors, register, handleSubmit } = useForm<AdminEditUserFormValues>({
+    resolver: joiResolver(AdminEditUserFormSchema),
+  });
+
+  async function onSubmit(
+    values: AdminEditUserFormValues,
+    event?: React.BaseSyntheticEvent
+  ): Promise<void> {
+    event?.preventDefault();
+    if (submitting) {
+      return;
+    }
+    try {
+      const response = await fetch(`/api/admin/users/${user?.id}`, {
+        method: "PATCH",
+        headers: { "Content-Type": "application/json" },
+        credentials: "include",
+        body: JSON.stringify({
+          email: values.email,
+          name: `${values.firstName} ${values.lastName}`,
+          phoneNumber: values.phoneNumber,
+        } as UpdateUserDTO),
+      });
+      if (!response.ok) {
+        throw await response.json();
+      }
+    } catch (err) {
+      setError(err.message);
+    } finally {
+      setSubmitting(false);
+      setIsEditing(false);
+    }
+  }
+  return (
+    <Modal open={Boolean(isEditing)}>
+      <div className="mx-16 mt-24">
+        <h1 className="text-3xl font-display font-medium mb-2">
+          Basic Information
+        </h1>
+        <form className="mt-10" onSubmit={handleSubmit(onSubmit)}>
+          <fieldset>
+            <FormField
+              label="First Name"
+              name="firstName"
+              error={errors.firstName?.message}
+            >
+              {user?.name ? (
+                <input
+                  type="text"
+                  className="input input-full"
+                  name="firstName"
```

Comment:
These fields share a lot in common with the admin/invite/new form. Wondering if there's a way we can reuse those bits (since @joyydong has also made changes to it in #42 that may be worth including in this form).

Reply:
sounds good! Will add in Joy's changes in my next PR